app/assets/javascripts/startpage/progressBar.coffee

```diff
@@ -0,0 +1,48 @@
+progressBar = angular.module("ProgressBar", ["StartPageServices"])
+
+progressBar.config(["$routeProvider", ($routeProvider) ->
+    $routeProvider.when("/progress_bar", {
+        templateUrl: "/assets/progress_bar.html",
+        controller: "ProgresBarController"
+    })
+])
+
+progressBar.controller("ProgressBarController", ["$scope", "StartPageStateData", ($scope, StartPageStateData) ->
+
+    currentID = StartPageStateData.currentTopic
+    percentComplete = 0
+
+    $scope.numQuestions = () -> StartPageStateData.numQuestions
+    $scope.questionsLeft = () -> StartPageStateData.questionsLeft
+    $scope.numTopics = () -> StartPageStateData.numTopics()
+    $scope.isTopicDone = () -> StartPageStateData.isTopicQuestionsDone(currentID)
+    $scope.topicsCompleted = () -> StartPageStateData.topicsCompleted
+
+    # %complete = (numCompleteTopics * 100/numTopics) + (numAnsweredQuestions * 100/totalQuestions * 100/numTopics)
+    $scope.percentComplete = () ->
+        updateCompletedTopics()
+
+        if (currentID == StartPageStateData.latestTopic)
+            numAnsweredQuestions = $scope.numQuestions() - StartPageStateData.questionsLeft
+        else
+            numAnsweredQuestions = $scope.numQuestions() - StartPageStateData.questionsLeft_static
```

Comment:
There are two problems with your approach here:
1) StartPageStatData.questionsLeft_static can be greater than $scope.numQuestions(), in which case currentPercent will be negative. 

2) Another problem is that the number of questions on the current topic is not necessarily the same as that of the latest topic, of which the questionsLeft_static is based on. This can cause the percentage of the progress bar to change when you only move to a previous questions page.